docs/gen-ai/gen-ai-events.md

```diff
@@ -0,0 +1,380 @@
+For example, when using OpenAI client libraries to communicate with Mistral, the `gen_ai.system`
+is set to `openai` based on the instrumentation's best knowledge.
+
+For custom model, a custom friendly name SHOULD be used.
+If none of these options apply, the `gen_ai.system` SHOULD be set to `_OTHER`.
+
+
+
+`gen_ai.system` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
+
+| Value  | Description | Stability |
+|---|---|---|
+| `anthropic` | Anthropic | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `cohere` | Cohere | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `openai` | OpenAI | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vertex_ai` | Vertex AI | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+<!-- END AUTOGENERATED TEXT -->
+<!-- endsemconv -->
+
+## System message
+
+This event describes the instructions passed to the GenAI model.
+It's NOT RECOMMENDED to report this event when capturing content is disabled, unless there are additional system-specific fields
+instrumentation adds to the event body.
+
+The event name MUST be `gen_ai.system.message`.
+
+| Body Field | Type | Description | Examples | Requirement Level |
+|---|---|---|---|---|
+| `content` | `AnyValue` | The contents of the system message. | `"You're a friendly bot that answers questions about OpenTelemetry."` | `Opt-In` |
+
+## User message
+
+This event describes the prompt message specified by the user.
+
+The event name MUST be `gen_ai.user.message`.
+
+| Body Field | Type | Description | Examples | Requirement Level |
+|---|---|---|---|---|
+| `content` | `AnyValue` | The contents of the user message. | `What telemetry is reported by OpenAI instrumentations?` | `Opt-In` |
+
+## Assistant message
+
+This event describes the assistant message.
+
+The event name MUST be `gen_ai.assistant.message`.
+
+| Body Field   | Type                           | Description                            | Examples                                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|--------------|--------------------------------|----------------------------------------|-------------------------------------------------|-------------------|
+| `content`    | `AnyValue`                     | The contents of the assistant message. | `Spans, events, metrics defined by the GenAI semantic conventions.` | `Opt-In` |
+| `tool_calls` | [ToolCall](#toolcall-object)[] | The tool calls generated by the model, such as function calls. | `[{"id":"call_mszuSIzqtI65i1wAUOE8w5H4", "function":{"name":"get_link_to_otel_semconv", "arguments":{"semconv":"gen_ai"}}, "type":"function"}]` | `Conditionally Required`: if available |
+
```

Comment:
(This is perhaps a question that could get sorted out later.)

I notice that Anthropic tool calls (they call them `tool_use`) are included in the array of `content`, e.g.:

https://docs.anthropic.com/en/docs/build-with-claude/tool-use#example-of-empty-tool-result
```json
{
  "id": "msg_01Aq9w938a90dw8q",
  "model": "claude-3-5-sonnet-20240620",
  "stop_reason": "tool_use",
  "role": "assistant",
  "content": [
    {
      "type": "text",
      "text": "<thinking>I need to use the get_weather, and the user wants SF, which is likely San Francisco, CA.</thinking>"
    },
    {
      "type": "tool_use",
      "id": "toolu_01A09q90qw90lq917835lq9",
      "name": "get_weather",
      "input": {"location": "San Francisco, CA", "unit": "celsius"}
    }
  ]
}
```

Should this section state a preference on how to handle that? Should the `content` be kept in full *and* the `tool_calls` duplicate some of that info?

Reply:
We should add a separate file for anthropic that would describe how all genai semconv would apply to anthropic. 

e.g. we use this pattern for dbs or messaging systems: 

https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/elasticsearch.md contains all you need to know about elasticsearch database spans/instrumentation.

Or https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/kafka.md documents what applies to kafka and adds some examples.

We could describe how exactly events are recorded for anthropic there, but I believe they fit nicely into the type system defined in this PR:

- choice event will contain both - the `content` and `tool_calls`: 

```json
{
  ...
  "message":  {
    "content": "<thinking>I need to call the get_link_to_otel_semconv function, and the user wants genai, which is likely GenAI.</thinking>",  
    "tool_calls":  [ { 
        "id":"toolu_01A09q90qw90lq917835lq9",
        "function":{"name":"get_link_to_otel_semconv","arguments":"{\"semconv\":\"GenAI\"}"},"type":"tool_use"
       }
     ]
  }
}
```

Based on docs, it seems it won't return more than one content block 
> When Claude decides to use one of the tools you’ve provided, it will return a response with a stop_reason of tool_use and one or more tool_use content blocks in the API response...

- when tool result it sent back to the model, we'll record `gen_ai.assistant.message` with the message object above and also the `gen_ai.tool.message` with the result of the tool:

```json
{"content":"opentelemetry.io/semconv/gen-ai","id":"toolu_01A09q90qw90lq917835lq9"}
```

If there are multiple tools, each response should be submitted as individual event.

The full example would look like this:

1. Chat completion resulting in a tool call.

   |   Attribute name    |                     Value                             |
   |---------------------|-------------------------------------------------------|
   | Span name           | `"chat claude-3-5"`                             |
   | `gen_ai.system`     | `"anthropic"`                                            |
   | `gen_ai.request.model`| `"claude-3-5`                                           |
   | `gen_ai.request.max_tokens`| `200`                                          |
   | `gen_ai.request.top_p`| `1.0`                                               |
   | `gen_ai.response.id`| `"msg_01Aq9w938a90dw8q"`            |
   | `gen_ai.response.model`| `"claude-3-5"`                                     |
   | `gen_ai.response.finish_reasons`| `["tool_calls"]`                          |

   Events parented to this span:

   - `gen_ai.user.message` 

     |   Property          |                     Value                             |
     |---------------------|-------------------------------------------------------|
     | Event body          | `{"content":"How to instrument antropic client with OTel?"}` |

   - `gen_ai.choice`

     |   Property          |                     Value                             |
     |---------------------|-------------------------------------------------------|
     | Event body    | `{"finish_reason":"tool_calls","message":{"content": "<thinking>I need to call the get_link_to_otel_semconv function, and the user wants antropic , which is likely Anthropic .</thinking>",  "tool_calls":[{"id":"toolu_01A09q90qw90lq917835lq9","function":{"name":"get_link_to_otel_semconv","arguments":"{\"semconv\":\"Anthropic\"}"},"type":"tool_use"}]}}` |

2. Application executes the tool call. Application may create span which is not covered by this semantic convention.
3. Final chat completion call

   |   Attribute name                |                     Value                             |
   |---------------------------------|-------------------------------------------------------|
   | Span name                       | `"chat claude-3-5"`                                        |
   | `gen_ai.system`                 | `"anthropic"`                                            |
   | `gen_ai.request.model`          | `"claude-3-5"`                                             |
   | `gen_ai.request.max_tokens`     | `200`                                                 |
   | `gen_ai.request.top_p`          | `1.0`                                                 |
   | `gen_ai.response.id`            | `"msg_01Aq9w938a90dw8q"`            |
   | `gen_ai.response.model`         | `"claude-3-5"`                                        |
   | `gen_ai.response.finish_reasons`| `["stop"]`                                            |

   Events parented to this span:
   (in this example, the event content matches the original messages, but applications may also drop messages or change their content)

   - `gen_ai.user.message` ...
   - `gen_ai.assistant.message`

     |   Property                       |                     Value                                                                                                                  |
     |----------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------|
     | Event body  | `{"content": "<thinking>I need ...", "tool_calls":[{"id":"toolu_01A09q90qw90lq917835lq9","function":{"name":"get_link_to_otel_semconv","arguments":"{\"semconv\":\"GenAI\"}"},"type":"tool_use"}]}` |

   - `gen_ai.tool.message`

     |   Property                       |                     Value                                                                      |
     |----------------------------------|------------------------------------------------------------------------------------------------|
     | Event body    | `{"content":"opentelemetry.io/semconv/gen-ai","id":"toolu_01A09q90qw90lq917835lq9"}` |

Reply:
Maybe I should rename `gen_ai.tool.message` to `gen_ai.tool_respose.message` or  `gen_ai.tool_result.message` so it's more obvious that it represents the outcome and not the call?